service/subresource_destroyer.go

```diff
@@ -44,66 +49,70 @@ func DeleteCascade(tenantId *int64, userId *int64, resourceType string, resource
 			return fmt.Errorf(`could not completely delete the source: %s`, err)
```

Comment:
@lindgrenj6 is it ok here when it fails here ? Should not we unlock mutex as well ?

Reply:
Yes, at this point we haven't launched any goroutines yet. The async part only really kicks off once the "parent" resource gets deleted, in this case the Source would have to completely succeed in deleting before launching the kafka producer routines.

Reply:
thanks!!